Blog/gst-build-cross-build/cross-building-with-gstreamer.md

```diff
@@ -0,0 +1,320 @@
+          gst-editing-services: YES
+                  gst-examples: YES
+    gst-integration-testsuites: YES
+                     gst-libav: YES
+                       gst-omx: YES
+               gst-plugins-bad: YES
+              gst-plugins-base: YES
+              gst-plugins-good: YES
+                gst-plugins-rs: NO
+              gst-plugins-ugly: YES
+                    gst-python: NO
+               gst-rtsp-server: YES
+                     gstreamer: YES
+               gstreamer-sharp: Feature 'sharp' disabled
+               gstreamer-vaapi: YES
+                         gtest: NO
+                   libmicrodns: YES
+                       libnice: YES
+                        libpsl: YES
+                       libsoup: NO
+                      openh264: YES
+                           orc: YES
+                     pygobject: NO
+                        sqlite: YES
+                      tinyalsa: NO
+                          x264: YES
+Option buildtype is: debug [default: debugoptimized]
+Found ninja-1.8.2 at /usr/bin/ninja
+
+```
+
+After this steps, a new folder `build_dir` should be ready to be used by `ninja` to build the binaries.
+
+##### Build gst-build
+
+This step will build all GStreamer base libraries in addition to the plugins from base/good/bad/ugly/libav if their
+dependencies have been met or built by `gst-build` (ie glib, openh264 etc.).
+
+```
+# ninja -C build_dir
+```
+
+
+#### Test gst-build
+
+This command will create a new environment where all tools and plugins built previously are available in the env super-setting the system one with the right environment variables.
+
+```
+# ninja -C build_dir uninstalled
+```
+
+A prefix to  your prompt should be available as
+
+
+```
+[gst-master] bash-prompt #
+```
+
+```
+[gst-master] bash-prompt # env | grep GST_
+```
+
+From this environment you are now ready to use the power of GStreamer, and even implement new features in it without the fear of using out of date version.
+
+From this shell, you are also able to compile without exiting the environment. This feature is very convenient to test a branch or fix a bug. Go to the `subprojects` folder and modify the code directly.
+
+```
+[gst-master] bash-prompt # gst-inspect-1.0
+```
+
+### Let's Talk about Cross-Compilation:
+
+As `gst-build` is the perfect partner to get started with a GStreamer development, here is my experience to perform a cross-build which can be very useful when you want to save precious build time or be able to work on both host and target with the same base code.
+
+In this post, I will target an **aarch64** CPU for the [Xilinx)(https://www.xilinx.com/) reference design: **Zynq UltraScale+ MPSoC rev F**
+
+#### Prerequisites
+
+* Tool-chain (aarch64-linux-gnu-gcc) + sysroot (optional)
+* Meson cross file
+* Meson > 0.54 (which is not available on blog creation date but [patch merged](https://github.com/mesonbuild/meson/pull/6461) in the official repository)
```

Comment:
Maybe wait for this feature to be released before publishing the post?

Reply:
yeah let see